src/CTA.Rules.Metrics/Models/CTAMetric.cs

```diff
@@ -0,0 +1,10 @@
+﻿using Newtonsoft.Json;
+
+namespace CTA.Rules.Metrics
+{
+    public abstract class CTAMetric
```

Comment:
why abstract ? you don't have any abstract method..

Reply:
This is abstract to prevent developers from instantiating this class directly. This serves as a starting template for all of our metrics classes, but this class itself should never be written instantiated.

[Here is an example of how we use this common base class to our advantage](https://github.com/aws/cta/blob/main/src/CTA.Rules.Metrics/PortSolutionResultReportGenerator.cs#L59). We aggregate all of our metrics objects into a single collection (using this class as the generic type) to generate a report.

Reply:
so you make them internal..

Reply:
Making the class internal would make sense here (it would also make sense to change all metrics object classes to internal), but within this assembly we still would want to avoid instantiating a `CTAMetric` metric directly.